rele/client.py

```diff
@@ -161,6 +165,14 @@ def publish(self, topic, data, blocking=False, timeout=None, **attrs):
         if not blocking:
             return future
 
-        future.result(timeout=timeout or self._timeout)
+        try:
+            future.result(timeout=timeout or self._timeout)
+        except TimeoutError as e:
+            run_middleware_hook("post_publish_failure", topic, e, data)
+            if raise_exception:
+                raise e
+            else:
+                return False
```

Comment:
What do you think about this to get rid of the `else`?
```suggestion
            if raise_exception:
                raise e
                
            return False
```